From 762655cadc32f36085c96016824e0610563b83ac Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Fri, 12 Aug 2016 12:55:40 +0700
Subject: [PATCH 38/42] gst145-Revert "omxvideodec: Set the already_acquired
 flag when the frame is dropped"

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

This reverts commit 86881767a6e5654ce587de80ac5e1f553c1192e1 at renesas-devel/gst-omx.git tag Yct_v1.10.0
which corresponding to the patch file for gst-omx 1.2.0:
	0012-omxbufferpool-avoid-the-duplicated-buffers-insertion.patch

Conflicts:
        omx/gstomxvideodec.c

This flag was considered to be tested whether it is set in the
release_buffer() method of GstBufferPool, but actually a buffer
dropped in this case will be freed with the method of GstOmxPort.

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxvideodec.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index f0f080d..efa0689 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -1599,21 +1599,11 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
   if (frame
       && (deadline = gst_video_decoder_get_max_decode_time
           (GST_VIDEO_DECODER (self), frame)) < 0) {
-    GstOMXVideoDecBufferData *vdbuf_data;
-
     GST_WARNING_OBJECT (self,
         "Frame is too late, dropping (deadline %" GST_TIME_FORMAT ")",
         GST_TIME_ARGS (-deadline));
     flow_ret = gst_video_decoder_drop_frame (GST_VIDEO_DECODER (self), frame);
     frame = NULL;
-    if (self->out_port_pool)
-    {
-      /* Sets the already_acquired flag a GstOMXBuffer buffer has
-       * in the case of the frame dropping, similar to the buffer acquisition.
-       */
-      vdbuf_data = (GstOMXVideoDecBufferData *) buf->private_data;
-      vdbuf_data->already_acquired = TRUE;
-    }
   } else if (!frame && (buf->omx_buf->nFilledLen > 0 || buf->eglimage)) {
     GstBuffer *outbuf = NULL;
 
-- 
1.9.1

